custom/icds/forms.py

```diff
@@ -228,14 +212,20 @@ def __init__(self, *args, **kwargs):
         )
 
     def clean_start_date(self):
-        data = self.cleaned_data['start_date']
-        validate_date(data)
-        return data
+        start_date = self.cleaned_data['start_date']
+        try:
+            start_date = parser.parse(start_date).date()
+        except ValueError:
+            raise forms.ValidationError(_("Invalid date"))
+        return start_date
 
     def clean_end_date(self):
         start_date = self.cleaned_data['start_date']
         end_date = self.cleaned_data['end_date']
-        validate_date(end_date)
-        if(start_date < end_date):
-            return end_date
+        try:
+            end_date = parser.parse(end_date).date()
+            if start_date <= end_date:
+                return end_date
+        except ValueError:
+            raise forms.ValidationError(_("Invalid date"))
```

Comment:
nit: This might be easier to follow
```
try:
    end_date = parser.parse(end_date).date()
except ValueError:
    raise forms.ValidationError(_("Invalid date"))
if start_date > end_date:
    raise forms.ValidationError(_("Start date cannot be greater than end date"))
return end_date
```